src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8753,31 +8706,28 @@ private TypeWithAnnotations GetDeclaredParameterResult(ParameterSymbol parameter
 
         public override BoundNode? VisitImplicitIndexerAccess(BoundImplicitIndexerAccess node)
         {
-            // The argument will be visited as part of VisitImplicitIndexerValuePlaceholder (for the first argument)
-            // to maintain proper order of evaluation
-            AddPlaceholder(node.ArgumentPlaceholders[0], node.Argument);
+            VisitRvalue(node.Receiver);
+            var receiverResult = _visitResult;
+            VisitRvalue(node.Argument);
+
+            EnsurePlaceholdersToResultMap();
+            _resultForPlaceholdersOpt.Add(node.ReceiverPlaceholder, (node.Receiver, receiverResult));
```

Comment:
Consider extracting to `NullableWalker.AddPlaceholderReplacement` and `RemovePlaceholderReplacement`

Reply:
> Consider extracting to NullableWalker.AddPlaceholderReplacement and RemovePlaceholderReplacement

I prefer to keep the churn to a minimum.